tests/uart_tests.py

```diff
@@ -0,0 +1,149 @@
+            self.skipTest("For real device only")
+
+        self.device.open()
+        uart : SupernovaUARTBlockingInterface = self.device.create_interface("uart") # Type hinting for easier development
+
+        (success, _) = uart.set_bus_voltage(3300)
+        uart.init_bus()
+
+        data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+        (success, _) = uart.send(data)
+        (successReceive, response) = uart.wait_for_notification(1)
+
+        self.assertEqual(success, True)
+        self.assertTupleEqual((successReceive, response), (True, data), f"Failed echo: {response}")
+
+        self.device.close()
+
+    def test_set_bus_voltage(self):
+        self.device.open()
+        uart : SupernovaUARTBlockingInterface = self.device.create_interface("uart") # Type hinting for easier development
+
+        (success, _) = uart.set_bus_voltage(3300)
+
+        self.assertEqual(success, True)
+
+        self.device.close()
+
+    def test_wait_timeout(self):
+        self.device.open()
+        uart : SupernovaUARTBlockingInterface = self.device.create_interface("uart") # Type hinting for easier development
+
+        uart.set_bus_voltage(3300)
+        uart.init_bus()
+
+        threading.Timer(5, lambda: self.fail("Timeout was exceeded"))
+        (success, response) = uart.wait_for_notification(4)
+        self.assertTupleEqual((False, "Timeout occurred while waiting for the UART receive notification"), (success, response), f"Failed non-timeout response: {response}")
+
+        self.device.close()
+
+    def test_uart_init_bus(self):
+        if self.use_simulator:
+            self.skipTest("For real device only")
+            # Currently there is a discrepancy in signature for the init bus methods between simulator and real SDK, making this test fail with the SIM
+
+        self.device.open()
+        uart : SupernovaUARTBlockingInterface = self.device.create_interface("uart") # Type hinting for easier development
+
+        (success, response) = uart.set_bus_voltage(330000)
+        self.assertEqual(False, success, f"Invalid excessive voltage set was successful: {response}")
+        (success, response) = uart.set_bus_voltage(3)
+        self.assertEqual(False, success, f"Invalid low voltage set was successful: {response}")
+
+        self.device.close()
+
+    # TODO Fails due to lack of management of ALREADY_INITIALIZED in the blocking interface
+    def test_uart_init_bus(self):
+        if self.use_simulator:
+            self.skipTest("For real device only")
+            # Currently there is a discrepancy in signature for the init bus methods between simulator and real SDK, making this test fail with the SIM
+
+        self.device.open()
+        uart : SupernovaUARTBlockingInterface = self.device.create_interface("uart") # Type hinting for easier development
+
+        uart.set_bus_voltage(3300)
+        (success, response) = uart.init_bus()
+
+        self.assertEqual(success, True, f"Failed: {response}")
```

Comment:
Why is the test failing because of an `ALREADY_INITIALIZED` error if uart init bus is performed only once?

Reply:
Since there is no way of checking if the bus initialized before, nor if we get the already_init_error or to close the bus, we cant really ensure if the init failed in a unexpected way or if it was started by a previous test